model/inline/src/main/scala/aqua/model/inline/MakeStructRawInliner.scala

```diff
@@ -25,18 +25,26 @@ import cats.syntax.apply.*
 
 object MakeStructRawInliner extends RawInliner[MakeStructRaw] {
 
-  private def createObj(fields: NonEmptyMap[String, ValueModel], result: VarModel): OpModel.Tree = {
-    val args = fields.toSortedMap.toList.flatMap { case (name, value) =>
+  private def createObj[S: Mangler](
+    fields: NonEmptyMap[String, ValueModel],
+    result: VarModel
+  ): State[S, OpModel.Tree] = {
+    fields.toSortedMap.toList.flatMap { case (name, value) =>
       LiteralModel.fromRaw(LiteralRaw.quote(name)) :: value :: Nil
+    }.map(TagInliner.canonicalizeIfStream(_, None)).sequence.map { argsWithOps =>
+      val (args, ops) = argsWithOps.unzip
+      val createOp =
+        CallServiceModel(
+          LiteralModel("\"json\"", ScalarType.string),
+          "obj",
+          CallModel(
+            args,
+            CallModel.Export(result.name, result.`type`) :: Nil
+          )
```

Comment:
Maybe move that pattern to a function so it's more readable? I see you do service calls like that in a few places. 

Something like this might be nice (pseudocode)
```scala
def callService(service: String, function: String, args: HList) -> CallServiceModel = {
  CallServiceModel(LiteralModel(s"\"$service\"", ScalarType.string), function, args)
}
```